zk-token-sdk/src/curve25519/scalar.rs

```diff
@@ -18,7 +18,7 @@ mod target_arch {
         type Error = Curve25519Error;
 
         fn try_from(pod: &PodScalar) -> Result<Self, Self::Error> {
-            Scalar::from_canonical_bytes(pod.0).ok_or(Curve25519Error::PodConversion)
+            Option::from(Scalar::from_canonical_bytes(pod.0)).ok_or(Curve25519Error::PodConversion)
```

Comment:
API like `Scalar::from_canonical_bytes` returns `CtOption` which doesn't have `ok_or` API. Do you have better ideas?